features/step_definitions/user_steps.rb

```diff
@@ -44,6 +44,8 @@ def user_agrees_to_membership_guidelines(user)
     AdminOnly::UserChecklistFactory.create_member_guidelines_checklist_for(new_user)
     user_agrees_to_membership_guidelines(new_user) if user_agreed_to_membership_guidelines
     new_user
+
+    new_user.start_membership!(send_email: false) if new_user.member?
```

Comment:
should really query the membership state: `.current_member?` 
AASM gem creates all of the query/boolean methods based on the different states.
That ensures that all of the requirements are checked and everything is up to date.
(Same as with the ShfApplication, of course)

Reply:
Yes, I know, but that method fails when called in the context of `problem-scenarios-pagination-view_all_companies`.  For purposes of that test, I just need the users to appear to be "members" (and that test (line 179) doesn't care that the members actually have not paid their membership fee, and thus cannot be members in the real world).

The set up data is: 

```
And the following users exist:
      | email         | admin | member | agreed_to_membership_guidelines |
      | u1@mutts.com  |       | true   | true                            |
      | u2@mutts.com  |       | true   | true                            |
...
```
where we're using the "member" attribute to indicate that the user should _look_ like a member for test purposes.

Short of restructuring many of our test setups around the actual requirements for an object to meet criteria for a state, I think this is a reasonable compromise, as long as mocking (in effect) the actual requirements doe not affect the integrity of the specific test.

Reply:
I've already done all that work to change the tests.  That was a big part of what I had to do to implement the new membership states, the Membership class, and the business logic.

 When someone new joins the project and reads through the tests, yours could (probably would) mislead them in understanding both the requirements and the implementation.

You need to:
1. set the `membership_status.  That is now how we track the membership status (since there is more than just the binary 'yes a member / no, not a member' state.  
2. create a 'Membership' object

You can see an example in the `features/user_account/member-home-account-page.feature`
There's a column for `membership_status` and then a section in the Background for a membership: `And the following memberships exist`  (I made sure to set this up so you can copy the first few columns from the payments set-up.)